src/mpid/ch4/netmod/ofi/ofi_recv.h

```diff
@@ -285,16 +285,6 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_irecv(void *buf,
                              MPI_ERR_OTHER, "**nomem", "**nomem %s", "Recv Pack Buffer alloc");
         recv_buf = MPIDI_OFI_REQUEST(rreq, noncontig.pack.pack_buffer);
         MPIDI_OFI_REQUEST(rreq, noncontig.pack.buf) = buf;
-#ifdef MPL_HAVE_ZE
-        if (dt_contig && attr.type == MPL_GPU_POINTER_DEV) {
-            int mpl_err = MPL_SUCCESS;
-            void *ptr;
-            mpl_err = MPL_ze_mmap_device_pointer(buf, &attr.device_attr, attr.device, &ptr);
-            MPIR_ERR_CHKANDJUMP(mpl_err != MPL_SUCCESS, mpi_errno, MPI_ERR_OTHER,
-                                "**mpl_ze_mmap_device_ptr");
-            MPIDI_OFI_REQUEST(rreq, noncontig.pack.buf) = ptr;
-        }
-#endif
```

Comment:
BTW, this mmap is unrelated to `MPL_gpu_fast_memcpy`, which is only used for small messages. If I recall, it is for improving performance of large messages in the case where GPU RDMA is slow. The mmap and change to `noncontig.pack.buf` causes a host copy instead of device copy. Maybe @zhenggb72 can clarify?

I think this particular case we will want to see the performance difference in case it causes degradation

Reply:
yes, I remember running osu_latency show significant difference in performance. Write performance is better than read, on the receive side, we write from host to device. If we do this mmap, we write from host, otherwise if launching a kernel on GPU, it is read. So there is performance difference. I don't think it should cause memory corruption or especially hang. But we may need to check if the caching works properly. I was more worrying about data integrity since  I think we may need to have a fence at the end of copying, but so far we never saw problems before.

Reply:
Adding this block back, the failures we see are from `test/mpi/coll/allgather_gpu 6` (and `allgatherv_gpu`):
```
not ok 30 - ././allgather_gpu 6 # time=1.85702610015869
  ---
  Directory: ./.
  File: allgather_gpu
  Num-procs: 6
  Timeout: 180
  Date: "Thu Oct  3 16:35:59 2024"
  ...
## Test output (expected 'No Errors'):
## /home/hzhou/MPI/bin/mpiexec -n 6    ./allgather_gpu -memtype=all
##  No Errors
## free(): invalid pointer
## free(): invalid pointer
## free(): invalid pointer
##
## ===================================================================================
## =   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
## =   PID 2263 RUNNING AT x1921c2s5b0n0.hostmgmt2000.cm.americas.sgi.com
## =   EXIT CODE: 6
## =   CLEANING UP REMAINING PROCESSES
## =   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
## ===================================================================================
## [proxy:0@x1921c2s3b0n0] HYD_pmcd_pmip_control_cmd_cb (proxy/pmip_cb.c:487): assert (!closed) failed
## [proxy:0@x1921c2s3b0n0] HYDT_dmxu_poll_wait_for_event (lib/tools/demux/demux_poll.c:76): callback returned error status
## [proxy:0@x1921c2s3b0n0] main (proxy/pmip.c:122): demux engine error waiting for event
## YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Aborted (signal 6)
## This typically refers to a problem with your application.
## Please see the FAQ page for debugging suggestions

```

The `free` messages are when we destroying `MPIDI_POSIX_global.per_vci[vci].am_hdr_buf_pool` in finalize. Apparently one of the slab was corrupted. I have not able to pinpoint where or how it gets corrupted. But removing the code in this commit fixes it.